indexer/src/main/scala/com/expedia/www/haystack/trace/indexer/writers/es/ElasticSearchWriter.scala

```diff
@@ -118,7 +120,8 @@ class ElasticSearchWriter(esConfig: ElasticSearchConfiguration, indexConf: White
     var isSemaphoreAcquired = false
 
     try {
-      addIndexOperation(traceId, packedSpanBuffer.protoObj, ElasticSearchWriterUtils.indexName(esConfig.indexNamePrefix, esConfig.indexHourBucket), isLastSpanBuffer) match {
+      val eventTimeInMicros = packedSpanBuffer.protoObj.getChildSpansList.asScala.map(_.getStartTime).fold(Long.MaxValue)(Math.min)
+      addIndexOperation(traceId, packedSpanBuffer.protoObj, ElasticSearchWriterUtils.indexName(esConfig.indexNamePrefix, esConfig.indexHourBucket, eventTimeInMicros), isLastSpanBuffer) match {
```

Comment:
minor comment, we can move indexName as separate variable to keep addIndexOperation short?
We can think of combining eventTimeInMicros and indexName as a separate function.
val indexName = indexName(packedSpanBuffer.protoObj);
addIndexOperation(traceId, .., indexName, ..)

Reply:
done